Complete rework

ffischer42 opened the pull request:
Merged all PEN related packages and updated PMT setup code & documentation.

ConnorHayward committed:
Merge pull request #2 from ffischer42/master

Merged all PEN related repositories

ffischer42 committed:
update 2021|04|08

ffischer42 committed:
Add files via upload

ffischer42 committed:
Update util.jl

ffischer42 committed:
Update util.jl

ffischer42 committed:
Removed some type definitions

e.g. WFSample caused problems while compiling
All functions should still be functional, but the type does not exist any more

ffischer42 committed:
Merge pull request #3 from mppmu/update

Update

ffischer42 committed:
Added variable threshold

ffischer42 committed:
added number_of_measurements to timing check

ffischer42 committed:
bug fix with "skip time" range

ffischer42 committed:
set number of measurements fix to 1

lmanzanillas committed:
check access

fhagemann committed:
Add functions to kill all java processes

fhagemann committed:
Retake failed measurements instead of saving them to missed_positions

fhagemann committed:
Always kill java processes that run longer than an hour

ffischer42 committed:
Merge pull request #4 from fhagemann/master

Update of PENBBScan2D
Killing left-over java processes after .scala script freezes. Status: Untested

ffischer42 committed:
Removed login payload

ffischer42 committed:
Changing time intervals

ffischer42 committed:
bbscan merge (manual)

Added move_to_ceph and escape if one point is not working

ffischer42 committed:
Update PENBBScan2D.jl

nVentis reviewed (approved):
Consider deleting previous versions of PENBBScan2D from git history

nVentis commented on src/struck/read_raw_data.jl:
```diff
@@ -35,53 +33,56 @@ function read_data_from_struck(filename::String; just_evt_t=false)
+    evt_t   = Float64[]
+    samples = Array{Int32,1}[]
+    chid    = Int32[]
+    energy  = []
```
Consider `energy = Float64[]`

nVentis commented on src/struck/read_raw_data.jl:
```diff
@@ -1,31 +1,29 @@
+    evt_t   = Float64[]
+    samples = Array{Int32,1}[]
+    chid    = Int32[]
+    energy  = []
```
energy explicit type

nVentis commented on src/struck/take_struck_data.jl:
```diff
@@ -14,41 +14,100 @@ Creates an individual `pmt_daq.scala` file and takes data which are converted to
-end
\ No newline at end of file
+    files = glob(joinpath(settings.data_dir, settings.output_basename * "*.dat"))
+    new_files = []
```
consider `String[]` (also below for h5files)

nVentis commented on src/util/util.jl:
```diff
@@ -4,19 +4,40 @@ log10orNaN(x) = (x <= 0) ? typeof(x)(NaN) : log10(x)
-"""
-function entrysel(predicate::Function, ds::DataFrame, colname::Symbol...)
+# function entrysel(predicate::Function, ds::DataFrame, colname::Symbol...)
+function entrysel(predicate, ds, colname)
```
Keep `predicate::Function`

Commit:
Merge branch 'master' of https://github.com/ffischer42/PENScintAnalysis.jl into ffischer42-master

nVentis committed:
Merge pull request #6 from mppmu/ffischer42-master

Ffischer42 master

Commit:
Removed merged dependency

Commit:
Updated for chunked data-taking

Commit:
Error handling

Commit:
Automatic process retries (max. 3 times total)

Commit:
More robust retries

Commit:
Cleaned up code

Commit:
Added more robust and quicker handling of timeouts

Commit:
Updated data processing, HDF5 conversion

Commit:
Merge branch 'master' of https://github.com/mppmu/PENScintAnalysis.jl

Commit:
Revert to updated take_struck_data

nVentis committed:
Debugging

nVentis committed:
Delete potentially wrongly recorded files
Removed debugging code

nVentis committed:
Updated error detection

nVentis committed:
Robust handling of Struck transfer errors:
- init error after 5000ms
- network error when retrying UDP package retrieval
Added examples for handled errors

nVentis committed:
Better error handling (functional)

nVentis committed:
Successfully tested
Adjusted parameters (sleep times etc)

nVentis committed:
Added measurement_state for chunked measurements
Added take_pmt_data to take measurements with one-line

nVentis committed:
Bug fixes

nVentis committed:
Final commits

nVentis committed:
Write data received before critical error

oschulz committed:
Remove Manifest.toml from version control

oschulz committed:
Add .gitignore

oschulz committed:
Remove obsolete Compat use

oschulz committed:
Update to current version of HTTP.jl
